packages/storage/src/reference.ts

```diff
@@ -203,42 +204,42 @@ export function getBlobInternal(
 export function getStreamInternal(
   ref: Reference,
   maxDownloadSizeBytes?: number
-): NodeJS.ReadableStream {
+): ReadableStream {
   ref._throwIfRoot('getStream');
-  const requestInfo = getBytes(
+  const requestInfo: RequestInfo<ReadableStream, ReadableStream> = getBytes(
     ref.storage,
     ref._location,
     maxDownloadSizeBytes
   );
 
-  /** A transformer that passes through the first n bytes. */
-  const newMaxSizeTransform: (n: number) => TransformOptions = n => {
+  // Transforms the stream so that only `maxDownloadSizeBytes` bytes are piped to the result
+  const newMaxSizeTransform = (n: number): Transformer => {
     let missingBytes = n;
     return {
-      transform(chunk, encoding, callback) {
+      transform(chunk, controller: TransformStreamDefaultController) {
         // GCS may not honor the Range header for small files
         if (chunk.length < missingBytes) {
-          this.push(chunk);
+          controller.enqueue(chunk);
           missingBytes -= chunk.length;
         } else {
-          this.push(chunk.slice(0, missingBytes));
-          this.emit('end');
+          controller.enqueue(chunk.slice(0, missingBytes));
+          controller.terminate();
         }
-        callback();
       }
-    } as TransformOptions;
+    };
   };
 
   const result =
     maxDownloadSizeBytes !== undefined
-      ? new Transform(newMaxSizeTransform(maxDownloadSizeBytes))
-      : new PassThrough();
+      ? new TransformStream(newMaxSizeTransform(maxDownloadSizeBytes))
+      : new TransformStream(); // The default transformer forwards all chunks to its readable side
 
   ref.storage
     .makeRequestWithTokens(requestInfo, newStreamConnection)
-    .then(stream => (stream as NodeJS.ReadableStream).pipe(result))
-    .catch(e => result.destroy(e));
-  return result;
+    .then(readableStream => readableStream.pipeThrough(result))
+    .catch(err => result.writable.getWriter().abort(err));
```

Comment:
This is all new to me so a couple of questions
(1) why `getWriter().abort(err)` and not `result.writable.abort(err)`? It looks like from the [docs](https://developer.mozilla.org/en-US/docs/Web/API/WritableStream#instance_methods) that `getWriter()` returns a new instance so we're creating a new one and then aborting it? I might be missing something because I don't know a lot about this streaming API.
(2) for both methods of `abort()` (the one on WritableStream and the one on WritableStreamDefaultWriter), it says the arg "reason" should be a string. I think most methods are smart enough to `.toString()` on errors but just wanted to double check this worked here

Reply:
> (1) why getWriter().abort(err) and not result.writable.abort(err)? It looks like from the [docs](https://developer.mozilla.org/en-US/docs/Web/API/WritableStream#instance_methods) that getWriter() returns a new instance so we're creating a new one and then aborting it? I might be missing something because I don't know a lot about this streaming API.

I believe the result is the same, but using `getWriter()` does seem like a roundabout way to `abort()`. I don't know why I hadn't noticed that `abort()` wasn't also on the `result.writable`. `getWriter()` does lock the stream, so maybe I thought it was better for that reason- but that's unnecessary here since we immediately abort anyway.

> (2) for both methods of abort() (the one on WritableStream and the one on WritableStreamDefaultWriter), it says the arg "reason" should be a string. I think most methods are smart enough to .toString() on errors but just wanted to double check this worked here

It does seem to interpret it correctly.

Here's a test that verifies the error thrown https://github.com/firebase/firebase-js-sdk/pull/8410/files#diff-c1e404f6a2985f11ee19789f49735e8fdbb51d16046f3c28f370433d92ca8431R78-R90

And in my sample app:
```ts
    const stream = getStream(ref(storage, 'asdfkasldkfjh'), 50); // Reference does not exist
    const reader = stream.getReader();
    reader.read().then(async function processImage({ done, value }) {
      if (done) {
        return;
      }

      result += value;
      reader.read().then(processImage);
    }).catch(err => console.error(err));
```
This is the output:
```
$ node src/index.js
[StorageError [FirebaseError]: Firebase Storage: Object 'asdfkasldkfjh' does not exist. (storage/object-not-found)] {
  code: 'storage/object-not-found',
  customData: { serverResponse: '' },
  status_: 0,
  _baseMessage: "Firebase Storage: Object 'asdfkasldkfjh' does not exist. (storage/object-not-found)"
}
```